datalake/src/main/java/com/sequenceiq/datalake/flow/salt/rotatepassword/handler/RotateSaltPasswordWaitHandler.java

```diff
@@ -0,0 +1,70 @@
+package com.sequenceiq.datalake.flow.salt.rotatepassword.handler;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.flow.salt.rotatepassword.event.RotateSaltPasswordFailureResponse;
+import com.sequenceiq.datalake.flow.salt.rotatepassword.event.RotateSaltPasswordSuccessResponse;
+import com.sequenceiq.datalake.flow.salt.rotatepassword.event.RotateSaltPasswordWaitRequest;
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+import com.sequenceiq.datalake.service.sdx.flowwait.SdxWaitService;
+import com.sequenceiq.datalake.service.sdx.flowwait.exception.SdxWaitException;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+import reactor.bus.Event;
+
+@Component
+public class RotateSaltPasswordWaitHandler extends ExceptionCatcherEventHandler<RotateSaltPasswordWaitRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RotateSaltPasswordWaitHandler.class);
+
+    @Value("${sdx.stack.rotatesaltpassword.sleeptime_sec}")
+    private int sleepTimeInSec;
+
+    @Value("${sdx.stack.rotatesaltpassword.duration_min}")
+    private int durationInMinutes;
+
+    @Inject
+    private SdxWaitService sdxWaitService;
+
+    @Inject
+    private SdxService sdxService;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(RotateSaltPasswordWaitRequest.class);
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<RotateSaltPasswordWaitRequest> event) {
+        LOGGER.warn("Fallback to default failure event for exception", e);
+        return new RotateSaltPasswordFailureResponse(resourceId, null, e);
+    }
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<RotateSaltPasswordWaitRequest> event) {
+        RotateSaltPasswordWaitRequest eventData = event.getData();
+        try {
+            SdxCluster sdxCluster = sdxService.getById(eventData.getResourceId());
+            PollingConfig pollingConfig = new PollingConfig(sleepTimeInSec, TimeUnit.SECONDS, durationInMinutes, TimeUnit.MINUTES)
+                    .withStopPollingIfExceptionOccurred(true);
+            sdxWaitService.waitForCloudbreakFlow(sdxCluster, pollingConfig, "Rotating salt password");
+            return new RotateSaltPasswordSuccessResponse(eventData.getResourceId(), event.getData().getUserId());
+        } catch (SdxWaitException e) {
+            LOGGER.warn("Error happened during waiting for salt password rotation {}, error: ",
```

Comment:
Nitpick: Errors usually "occur" and not "happen". E.g. "An error occurred during...."

Reply:
It was copypaste, we have 82 matches for "Error happened" in CB services :smile: But I can fix it here

Reply:
If you don't have anything else to fix, feel free to leave it as it is. It's a minor nitpick after all.